engine/src/test/java/org/hibernate/search/engine/cfg/ConfigurationPropertySourceTest.java

```diff
@@ -0,0 +1,323 @@
+					}
+
+					if ( scope.matchExact( "index", "my-index" ) ) {
+						return Optional.of( fromMap( singletonMap( "someOtherKey", "index-my-index-bar" ) ) );
+					}
+
+					if ( scope.matchExact( "index" ) ) {
+						return Optional.of( fromMap( singletonMap( "someOtherKey", "index-bar" ) ) );
+					}
+
+					return Optional.empty();
+				}
+		);
+		ConfigurationPropertySource propertySource = fromMap( asMap( "hibernate.search.someKey", "foo" ) )
+				.withMask( "hibernate" )
+				.withMask( "search" )
+				.withFallback( fallback( resolver, global() ) );
+
+		assertPair( propertySource, "foo", "global-bar" );
+
+		propertySource = propertySource.withFallback( fallback( resolver(), backend() ) );
+		assertPair( propertySource, "foo", "global-bar" );
+	}
+
+	@Test
+	public void multipleProviders() {
+		class ConfigurationProviderA implements ConfigurationProvider {
+
+			@Override
+			public Optional<ConfigurationPropertySource> get(ConfigurationScope scope) {
+				if ( scope.matchExact( "global" ) ) {
+					return Optional
+							.of( fromMap( singletonMap( "prefix.key", "global-value-a" ) ) );
+				}
+				return Optional.empty();
+			}
+
+			@Override
+			public String toString() {
+				return "ConfigurationProviderA";
+			}
+		}
+
+		class ConfigurationProviderB implements ConfigurationProvider {
+
+			@Override
+			public Optional<ConfigurationPropertySource> get(ConfigurationScope scope) {
+				if ( scope.matchExact( "global" ) ) {
+					return Optional
+							.of( fromMap( singletonMap( "prefix.key", "global-value-b" ) ) );
+				}
+				return Optional.empty();
+			}
+
+			@Override
+			public String toString() {
+				return "ConfigurationProviderB";
+			}
+		}
+
+		logged.expectEvent(
+				Level.DEBUG,
+				"Multiple configuration providers are available for scope 'global'.",
+				"They will be taken under consideration in such priority: '[ConfigurationProviderA, ConfigurationProviderB]'."
+		);
+		ConfigurationPropertySource propertySource = fromMap( asMap(
+				"prefix.key2", "value"
+		) ).withFallback(
+				fallback( resolver( new ConfigurationProviderB(), new ConfigurationProviderA() ), global() )
+		);
+
+		assertThat( propertySource.get( "prefix.key2" ) )
+				.isPresent()
+				.get()
+				.isEqualTo( "value" );
+		assertThat( propertySource.get( "prefix.key" ) )
+				.isPresent()
+				.get()
+				.isEqualTo( "global-value-a" );
+	}
+
+	@Test
+	public void almostRealisticScenario() {
+		BeanResolver resolver = resolver(
+				scope -> {
+					if ( scope.matchExact( "global" ) ) {
+						return Optional.of( fromMap( asMap(
+								"hibernate.search.backend.my-backend.index.my-index.override-setting", "global",
+								"hibernate.search.global-specific-setting", "global"
+						) ) );
+					}
+					if ( scope.matchExact( "backend", "my-backend" ) ) {
+						return Optional.of( fromMap( asMap(
+								"index.my-index.override-setting", "my-backend",
+								"my-backend-specific-setting", "my-backend"
+						) ) );
```

Comment:
Sure. But since I already had a test for a more abstract scenario from the previous version where it was overriding properties, I've updated it to test that it doesn't work, so we still test what we are stating as something that won't work 😃